src/server/async_disconnect_handler.cpp

```diff
@@ -50,9 +50,9 @@ template <typename T_CONN_HANDLER>
 void
 async_disconnect_handler<T_CONN_HANDLER>::disconnect (connection_handler_uptr_t &&handler)
 {
+  std::unique_lock<std::mutex> ulock { m_queue_mtx };
   if (!m_terminate.load ())
     {
-      std::unique_lock<std::mutex> ulock { m_queue_mtx };
```

Comment:
To avoid a race condition. With the previous code, the `m_terminate` can be true after the lock it released. It means it could push the disconnect handler after terminating.

Reply:
fair enough